src/main/java/io/vertx/core/VertxThreadFactoryCreator.java

```diff
@@ -0,0 +1,25 @@
+package io.vertx.core;
+
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.core.impl.BlockedThreadChecker;
+import io.vertx.core.impl.VertxThreadFactory;
+
+/**
+ * Extension point to allow creating custom {@code VertxThreadFactory} instances.
+ */
+public interface VertxThreadFactoryCreator {
```

Comment:
can you rename to `VertxThreadFactoryProvider`, we already have similar plugins that are named this way (like `ResolverProvider`), I think also it's a better name.

also it should be in the package `io.vertx.core.spi` instead